src/NeoSharp.Core/Persistence/ITransactionRepository.cs

```diff
@@ -0,0 +1,8 @@
+﻿using NeoSharp.Core.Models;
+
+namespace NeoSharp.Core.Persistence
+{
+    public interface ITransactionRepository : IRepository<Transaction>
+    {
```

Comment:
Maybe we need a `enum` for split all hashes on the storage, on the current version of neo, we have preffix for this hashes, it depends the Repository of course, in one could be tables, and in other key/value, but if IRepository constructor have the type of this repository, could help to store the data on the right place.

https://github.com/neo-project/neo/blob/master/neo/Implementations/Blockchains/LevelDB/DataEntryPrefix.cs

Reply:
yes ... this could in considered too
this was just an example and a lot of possible features were left out in order of simplicity